services/meta/client.go

```diff
@@ -71,7 +71,12 @@ func (c *Client) Close() error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	close(c.closing)
+	select {
+	case <-c.closing:
+		return nil
+	default:
```

Comment:
ahh, I see what you are doing.  If it isn't closed, you are closing it.  This is instead of setting the close channel to `nil`.  I like it.  Allows for multiple calls to close without blowing up.